lib/dogma/rules/literal_in_interpolation.ex

```diff
@@ -0,0 +1,50 @@
+defmodule Dogma.Rules.LiteralInInterpolation do
+  @moduledoc """
+  A rule that disallows useless string interpolations
+  that contain a literal value instead of a variable or function.
+  """
+
+  @behaviour Dogma.Rule
+
+  alias Dogma.Script
+  alias Dogma.Error
+
+  import Dogma.Util.ASTNode, only: [literal?: 1]
+
+  def test(script, _config = [] \\ []) do
+    script |> Script.walk( &check_node(&1, &2) )
+  end
+
+  defp check_node({:::, meta, args} = node, errors) do
```

Comment:
So one pattern Jose matches with is [here](https://github.com/elixir-lang/elixir/blob/aa901b1f5ad0749fecaf1fe5d89bbabcabf62399/lib/elixir/lib/macro.ex#L632). So `:<<>> ` is the salient function and arguments to that function are string literals and interpolations whose pattern is 
```
{:::, _,
   [{{:., _, [Kernel, :to_string]}, [], [(thing getting interpolated)]}, {:binary, _, _}]}}
```

`:::` by itself definitely over matches which is the reason for the error handling below. I originally had the whole pattern in the function def but that's pretty long and unwieldy. I'm sure you have more experience figuring out what to match on, open to suggestions.